### PR TITLE
Throw error when domain is set as function for non-…

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [x] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

Accessing this.domain inside Isomorphic clerk will result to runtime error when the window is undefined. 

For non-browser environments, prohibit the usage of a function for now.

### CODE REVIEW

File: packages/react/src/errors.ts [hunk 1]
```diff
@@ -31,3 +31,6 @@ export function isMagicLinkError(err: any): err is MagicLinkError {
 
 export const invalidStateError =
   'Invalid state. Feel free to submit a bug or reach out to support here: https://clerk.com/support';
+
+export const domainAsFunctionError =
+  'The usage of a function as domain is not supported in non-browser environments. Please contact support@clerk.com';
```
Comment (gkats) [inline packages/react/src/errors.ts:36]: :upside_down_face: Perhaps the "contact" part of the message is redundant?

File: packages/react/src/isomorphicClerk.ts [hunk 2]
```diff
@@ -92,6 +92,16 @@ export default class IsomorphicClerk {
   }
 
   get domain(): string {
+    /**
+     * This getter can run in environments where window is not available.
+     * In those cases we should expect and use domain a string
```
Comment (gkats) [inline packages/react/src/isomorphicClerk.ts:97]: :upside_down_face: 
```suggestion
     * In those cases we should expect and use domain as a string
```

**Review by gkats — APPROVED** [review]
:bug:  

Let's also add a commit message with more information on why this change is needed and if possible reproduce it in a test.

File: packages/react/src/isomorphicClerk.ts (hunk 2 above)
Comment (panteliselef) [reply packages/react/src/isomorphicClerk.ts]: Nice catch 😀

File: packages/react/src/errors.ts (hunk 1 above)
Comment (panteliselef) [reply packages/react/src/errors.ts:36]: TBH this was requested, but I may have taken it a bit seriously 
cc @nikosdouvlis

Comment (nikosdouvlis) [reply packages/react/src/errors.ts:36]: I suggested this error message as apart from the components, this is one of the few features we don't support on non-browser environment for now. This  error does not indicate missuse of our apis or an actual error - just a missing feature :)

Feel free to adjust accordingly or remove completely - no real issues here!

**Review by nikosdouvlis — APPROVED** [review]

File: packages/react/src/errors.ts (hunk 1 above)
Comment (panteliselef) [reply packages/react/src/errors.ts:36]: I am thinking of changing this one to 
```typescript
export const unsupporterDomainFunctionInNonBrowser =
  'Unsupported usage of domain. The usage of domain as function is not supported in non-browser environments.';
```

Comment (nikosdouvlis) [reply packages/react/src/errors.ts:36]: This is better, thanks Panteli

**Pull request merged**